### PR TITLE
Add validation for hub guidance

### PR DESCRIPTION
**Pull request by LJBabbage**
### PR Context
The schema can now be used to override the guidance message on the hub. I have also extracted the hub functionality into its own area, I felt the questionnaire_v1.json was getting a bit cumbersome and although the new guidance messages could feasibly go in common_definitions it also was on the last buckle of the belt. The hub is starting to feel like its own thing so I thought it justified the split, although I am not precious if someone wants to stick it somewhere else.

### CODE REVIEW

File: schemas/hub/definitions.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        "description": "Override guidance messages for the hub.",
+        "properties": {
+          "COMPLETE": {
+            "$ref": "https://eq.ons.gov.uk/hub/definitions.json#/messages"
+          },
+          "INCOMPLETE": {
+            "$ref": "https://eq.ons.gov.uk/hub/definitions.json#/messages"
+          }
+        },
+        "additionalProperties": false
```
Comment (MebinAbraham) [inline schemas/hub/definitions.json:21]: This should have:
```json
        "anyOf": [
          {
            "required": ["COMPLETE"]
          },
          {
            "required": ["INCOMPLETE"]
          }
        ],
```

File: schemas/hub/definitions.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        "type": "string"
+      },
+      "submit_button": {
+        "type": "string"
+      },
+      "description": {
+        "type": "string"
+      }
+    },
+    "additionalProperties": false
```
[the remark this answers is not in the document]
Comment (LJBabbage) [reply schemas/hub/definitions.json:47]: Makes sense

File: schemas/hub/definitions.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        "type": "boolean"
+      },
+      "guidance": {
+        "type": "object",
+        "description": "Override guidance messages for the hub.",
+        "properties": {
+          "COMPLETE": {
+            "$ref": "https://eq.ons.gov.uk/hub/definitions.json#/messages"
+          },
+          "INCOMPLETE": {
```
[the remark this answers is not in the document]
Comment (LJBabbage) [reply schemas/hub/definitions.json:17]: I was matching what we currently have in hub context and this was the structure

File: schemas/hub/definitions.json [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  },
+  "messages": {
+    "type": "object",
+    "properties": {
+      "title": {
+        "type": "string"
+      },
+      "submit_button": {
+        "type": "string"
+      },
```
[the remark this answers is not in the document]
Comment (LJBabbage) [reply schemas/hub/definitions.json:42]: True, but that wasn't in story description and why it is like it is "Changes to hub - make text configurable"

File: schemas/common_definitions.json [hunk 5]
```diff
@@ -297,5 +297,15 @@
       "additionalProperties": false,
       "required": ["goto"]
     }
+  },
+  "submission": {
+    "type": "object",
+    "properties": {
+      "button": {
```
Comment (ajmaddaford) [inline schemas/common_definitions.json:304]: I realise this was the structure I suggested, but as there is no submission title property, we should consider flattening this to `submit_button`.

File: schemas/common_definitions.json [hunk 6]
```diff
@@ -297,5 +297,15 @@
...
+        "type": "string",
+        "description": "The text that appears on the submit button, when the questionnaire is ready to be submitted"
+      }
+    },
+    "additionalProperties": false
```
Comment (ajmaddaford) [inline schemas/common_definitions.json:309]: This definition is specific to the hub so should be in `schemas/hub/definitions.json`.

File: schemas/hub/definitions.json [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        ]
+      },
+      "incomplete": {
+        "type": "object",
+        "description": "The guidance that will appear on the hub, when the questionnaire is incomplete.",
+        "properties": {
+          "guidance": {
+            "type": "string"
+          }
+        },
```
Comment (MebinAbraham) [inline schemas/hub/definitions.json:36]: The `title` and `guidance` keys should be defined as properties outside these `complete/incomplete` keys then referenced by them. It should also be mandatory to provide one of those keys.

File: schemas/hub/definitions.json [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        "additionalProperties": false
+      },
+      "submission": {
+        "type": "object",
+        "properties": {
+          "button": {
+            "type": "string",
+            "description": "The text that appears on the submit button, when the questionnaire is ready to be submitted"
+          }
```
Comment (MebinAbraham) [inline schemas/hub/definitions.json:45]: Should be made mandatory when `submission` is provided.

File: schemas/hub/definitions.json [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+        "items": {
+          "type": "string"
+        }
+      },
+      "complete": {
+        "type": "object",
+        "properties": {
+          "title": {
+            "type": "string",
+            "description": "The title that will appear on the hub, when the questionnaire is incomplete."
```
Comment (MebinAbraham) [inline schemas/hub/definitions.json:22]: `is complete`?

File: schemas/hub/definitions.json [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+          },
+          "guidance": {
+            "$ref": "https://eq.ons.gov.uk/hub/definitions.json#/guidance"
+          }
+        },
+        "anyOf": [
```
Comment (MebinAbraham) [inline schemas/hub/definitions.json:28]: Needs `"additionalProperties": false,`

**Comment by MebinAbraham** [comment]
Add do not merge until runner PR is ready to merge.

**Comment by MebinAbraham** [comment]
> Add do not merge until runner PR is ready to merge.

New properties are not mandatory.

**Pull request merged**